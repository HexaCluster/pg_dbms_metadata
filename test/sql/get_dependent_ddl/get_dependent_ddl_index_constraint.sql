SELECT dbms_metadata.get_dependent_ddl('INDEX','table_all','gdmmm');

SELECT dbms_metadata.get_dependent_ddl('INDEX','Sample unlogged table','gdmmm');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','table_all','gdmmm');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','table_all_child','gdmmm');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','Sample unlogged table','gdmmm');

SELECT dbms_metadata.get_dependent_ddl('REF_CONSTRAINT','table_all_child','gdmmm');

-- tests for schema as NULL
SET search_path TO public, gdmmm;

SELECT dbms_metadata.get_dependent_ddl('INDEX','table_all');

SELECT dbms_metadata.get_dependent_ddl('INDEX','Sample unlogged table');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','table_all');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','table_all_child');

SELECT dbms_metadata.get_dependent_ddl('CONSTRAINT','Sample unlogged table');
