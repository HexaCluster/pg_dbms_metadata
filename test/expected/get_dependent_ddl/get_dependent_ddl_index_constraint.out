                                     get_dependent_ddl                                     
-------------------------------------------------------------------------------------------
 CREATE INDEX perf_index ON gdmmm.table_all USING btree (c) TABLESPACE :"TABLESPACE_INDEX"+
 
(1 row)

                                                  get_dependent_ddl                                                   
----------------------------------------------------------------------------------------------------------------------
 ALTER TABLE gdmmm.table_all ADD CONSTRAINT table_all_pkey PRIMARY KEY (a) USING INDEX TABLESPACE :"TABLESPACE_INDEX"+
 ALTER TABLE gdmmm.table_all ADD CONSTRAINT table_all_d_key UNIQUE (d) USING INDEX TABLESPACE :"TABLESPACE_INDEX"    +
 
(1 row)

                                                 get_dependent_ddl                                                  
--------------------------------------------------------------------------------------------------------------------
 ALTER TABLE gdmmm.table_all_child ADD CONSTRAINT child_uniq UNIQUE (id) USING INDEX TABLESPACE :"TABLESPACE_INDEX"+
 
(1 row)

                                                                   get_dependent_ddl                                                                    
--------------------------------------------------------------------------------------------------------------------------------------------------------
 ALTER TABLE gdmmm.table_all_child ADD CONSTRAINT fk_customer FOREIGN KEY (id) REFERENCES gdmmm.table_all(a) USING INDEX TABLESPACE :"TABLESPACE_INDEX"+
 
(1 row)

